src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,282 @@
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  options?: FlickityOptions
+}
+
+const isClient = typeof window !== "undefined"
+
+/*
+ * Flickity docs https://flickity.metafizzy.co
+ * Flickity requires window, so we must render a seperate Flex wrapper instead of
+ * a carousel if we're using SSR.
+ */
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+  state = {
+    isMounted: false,
+  }
+
+  componentDidMount() {
+    this.setState({
+      isMounted: true,
+    })
+  }
+
+  /*
+   * If we don't check for isMounted, React will not properly clean up the
+   * CarouselServer component on initial render
+   */
+  render() {
+    return (
+      <>
+        {isClient && this.state.isMounted && (
+          <Box key="client">
+            <Media at="xs">
+              <SmallCarousel {...this.props} />
+            </Media>
+            <Media greaterThan="xs">
+              <LargeCarousel {...this.props} />
+            </Media>
+          </Box>
+        )}
+        {!isClient && !this.state.isMounted && (
+          <Box key="server">
+            <CarouselServer {...this.props} />
+          </Box>
+        )}
+      </>
+    )
+  }
+}
+
+const renderSlides = props => {
+  return props.data.map((slide, index) => {
+    return <Box key={index}>{props.render(slide)}</Box>
+  })
+}
+
+/*
+ * This version of the carousel is only rendered for SSR
+ */
+const CarouselServer = (props: Props) => {
+  return (
+    <Flex
+      flexDirection="row"
+      position="relative"
+      justifyContent="space-around"
+      alignItems="center"
+      height={props.height}
+    >
+      <CarouselContainer height={props.height}>
+        <Flex>{renderSlides(props)}</Flex>
+      </CarouselContainer>
+    </Flex>
+  )
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity = isClient ? require("react-flickity-component") : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+  const flicktyRef = useRef(null)
+  const isMounted = useDidMount()
+  let flickity = null
+
+  const checkLastItemVisible = () => {
+    if (!flickity.slides.length) {
+      return
+    }
+    const isLastItemVisible = flickity.selectedElements.includes(
+      flickity.getLastCell().element
+    )
+    setLastItemVisible(isLastItemVisible)
+  }
+
+  /*
+   * We need to wait for the component to mount before using the flickity ref.
+   * Flickity will only initialize after the component has rendered.
+   */
+  if (isMounted) {
+    if (flicktyRef.current !== null) {
+      flickity = flicktyRef.current.flkty
```

Comment:
`flickity` !== `flkty` -- better to always keep naming 1:1 as that reduces refactor burdens later on.

Reply:
I renamed the `flicktyRef` to `flickityRef`, Flickity uses `flkty` a lot in their docs and this is the object from the library which uses `flckty` here

Reply:
However it's used in the docs is how it should be referenced in local assignments, so 1:1.